loadbalancer/servicenode.go

```diff
@@ -0,0 +1,188 @@
+package loadbalancer
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/filecoin-project/boost/loadbalancer/messages"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/protocol"
+)
+
+// ServiceNode is a host that behaves as a service node connected to a load balancer
+// -- all traffic is routed through the load balancer for each registered protocol
+type ServiceNode struct {
```

Comment:
I'd suggest renaming ServiceNode to something more specific, like "ForwardingHost"